mesheryctl/internal/cli/root/system/check.go

```diff
@@ -42,7 +44,50 @@ var checkCmd = &cobra.Command{
 func RunPreflightHealthChecks(isPreRunExecution bool, subcommand string) error {
 	mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
 	if err != nil {
-		return errors.Wrap(err, "error processing config")
+		userResponse := false
+
+		//skip asking confirmation if -y flag used or host in meshconfig is already localhost
+		if utils.SilentFlag {
+			userResponse = true
+		} else {
+			// ask user for confirmation
+			userResponse = utils.AskForConfirmation("Do you want to create a new config file")
+		}
+
+		if !userResponse {
+			return err
+		}
+
+		cfgFile := utils.CfgFile
+		log.Println("Backing up " + cfgFile + " to original_config.yaml")
+		err := os.Rename(cfgFile, "original_config.yaml")
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		_, err = os.Create(cfgFile)
```

Comment:
Should we be doing this in root.go? i.e when the config file is being created? It will then run for each command and would not require check being invoked.

Reply:
Unless root.go is setup to run through a number of checks already, which are defined in a single location - `system check`....